### PR TITLE
Add static calculation feature to matcalc/relaxation.py

### PR DESCRIPTION
**Pull request by rul048**
In the original relaxation.py, users are allowed to do cell relaxation or atoms only relaxation. Static calculation is an important feature itself and also part of some complex applications (eg. QHA). As MatCalc provides a universal interface for different uMLIPs, how about we add this feature to RelaxCalc? The default is still cell relaxation, and user can set "relax_atoms" and "relax_cell" to False to do static calculation.

## Summary

Major changes:

- feature 1: ...
- fix 1: ...

## Todos

If this is work in progress, what else needs to be done?

- feature 2: ...
- fix 2:

## Checklist

- [x] Google format doc strings added. Check with `ruff`.
- [x] Type annotations included. Check with `mypy`.
- [x] Tests added for new features/fixes.
- [ ] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

Tip: Install `pre-commit` hooks to auto-check types and linting before every commit:

```sh
pip install -U pre-commit
pre-commit install
```

### CODE REVIEW

**Comment by rul048** [comment]
Another reason is that I will also upload a workflow for qha phonon calculations where static calculation is needed and I would not like to define it in that.

**Comment by shyuep** [comment]
This is good, but I will need unittests for all three different options.

**Comment by rul048** [comment]
Okay, I think it should be fine now. Credit to Dr Ko, learn a lot about how to do unittests from MatGL.

File: tests/test_relaxation.py [hunk 1]
```diff
@@ -35,8 +37,48 @@ def test_relax_calc_single(
...
+    relax_calc = RelaxCalc(
+        M3GNetCalc, traj_file=f"{tmp_path}/li2o_relax.txt", relax_atoms=False, relax_cell=False
+    )
+    result = relax_calc.calc(Li2O)
+
+    energy: float = result["energy"]
+    forces: ArrayLike = result["forces"]
+    stresses: ArrayLike = result["stress"]
+
+    assert isinstance(energy, float)
```
Comment (shyuep) [inline tests/test_relaxation.py:78]: These asserts should be based on value, not type.

**Pull request merged**